packages/core/src/client.ts

```diff
@@ -648,18 +536,21 @@ export class SkipRouter {
     );
 
     if (!accountFromSigner) {
-      throw new Error("Failed to retrieve account from signer");
+      throw new Error(
+        "signMultiChainMessageDirect error: failed to retrieve account from signer",
+      );
     }
 
     const message = getEncodeObjectFromMultiChainMessage(multiChainMessage);
 
-    const pubkey = encodePubkey(
-      encodeSecp256k1Pubkey(accountFromSigner.pubkey),
-    );
+    const algo = `${accountFromSigner.algo}`;
+    const isEthSecp256k1 = algo === "eth_secp256k1" || algo === "ethsecp256k1";
 
-    if ((accountFromSigner.algo as string) === "ethsecp256k1") {
-      pubkey.typeUrl = "/ethermint.crypto.v1.ethsecp256k1.PubKey";
-    }
+    const pubKey = isEthSecp256k1
+      ? encodeEthSecp256k1Pubkey(accountFromSigner.pubkey)
+      : encodeSecp256k1Pubkey(accountFromSigner.pubkey);
+
+    const encodedPubKey = encodePubkey(pubKey);
```

Comment:
We should probably break this logic into its own function, since 1) its pretty complex & 2) its used identically again on 746

Reply:
Hard agree, will extract in next commit.

Reply:
Impl in 0c7a27a7e6a83c0367ebe5f3d80463412ed558f8